addons/mrp/models/mrp_document.py

```diff
@@ -15,6 +15,13 @@ class MrpDocument(models.Model):
     }
     _order = "priority desc, id desc"
 
+    def copy(self, default=None):
+        ir_default = default
+        if ir_default:
+            ir_default = {key : default[key] for key in default.keys() if key in self.ir_attachment_id.fields_get()}
```

Comment:
```suggestion
            ir_fields = list(self.env['ir.attachment']._fields)
            ir_default = {field : default[field] for field in ir_fields if key in ir_fields}
```
to avoid a few (7) sql queries